### PR TITLE
8300806: Update googletest to v1.13.0

### PR DESCRIPTION
**Pull request by vidmik**
Please review this PR which updates the required version of googletest to v1.13.0.

It’s been (quite) a while since we upgraded the version of googletest used to run the hotspot unit tests (test/hotspot/gtest). The current version (1.8.1) is from 2018 and there is build issue with Xcode 14.x which has been resolved in more recent versions of googletest.

The configure.ac file has been removed from googletest so the only file which contains the version is CMakeLists.txt, hence the removed logic from `make/autoconf/lib-tests.m4`.

Testing: tier1, gtest:all, GHA

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8300806](https://bugs.openjdk.org/browse/JDK-8300806): Update googletest to v1.13.0

### Reviewers
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**)
 * [Magnus Ihse Bursie](https://openjdk.org/census#ihse) (@magicus - **Reviewer**) ⚠️ Review applies to [d037905b](https://git.openjdk.org/jdk/pull/12125/files/d037905bd838da6c6afe9a1d321252a283121882)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12125/head:pull/12125` \
`$ git checkout pull/12125`

Update a local copy of the PR: \
`$ git checkout pull/12125` \
`$ git pull https://git.openjdk.org/jdk pull/12125/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12125`

View PR using the GUI difftool: \
`$ git pr show -t 12125`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12125.diff">https://git.openjdk.org/jdk/pull/12125.diff</a>

</details>

### CODE REVIEW

File: doc/building.md [hunk 1]
```diff
@@ -867,10 +867,10 @@ Download the latest `.tar.gz` file, unpack it, and point `--with-jtreg` to the
 Building of Hotspot Gtest suite requires the source code of Google Test framework.
 The top directory, which contains both `googletest` and `googlemock`
 directories, should be specified via `--with-gtest`.
-The supported version of Google Test is 1.8.1, whose source code can be obtained:
+The supported version of Google Test is 1.13.0, whose source code can be obtained:
 
- * by downloading and unpacking the source bundle from [here](https://github.com/google/googletest/releases/tag/release-1.8.1)
- * or by checking out `release-1.8.1` tag of `googletest` project: `git clone -b release-1.8.1 https://github.com/google/googletest`
+ * by downloading and unpacking the source bundle from [here](https://github.com/google/googletest/releases/tag/v1.13.0)
+ * or by checking out `v1.13.0` tag of `googletest` project: `git clone -b v1.13.0 https://github.com/google/googletest`
```
Comment (erikj79) [inline doc/building.md:873]: You need to generate the html file for these changes. `make update-build-docs`

**Review by erikj79 — COMMENTED** [review]
Good to see we didn't need more changes in tests to make this work.

File: doc/building.md (hunk 1 above)
Comment (vidmik) [reply doc/building.md:873]: Of course, thank you!

**Review by erikj79 — APPROVED** [review]

**Review by magicus — APPROVED** [review]
Other than that, it looks good.

File: make/autoconf/lib-tests.m4 [hunk 2]
```diff
@@ -58,20 +58,10 @@ AC_DEFUN_ONCE([LIB_TESTS_SETUP_GTEST],
...
-          AC_MSG_ERROR([gtest at $GTEST_FRAMEWORK_SRC does not seem to be version 1.8.1])
-        fi
-
-        # We cannot grep for "AC_IN*T" as a literal since then m4 will treat it as a macro
-        # and expand it.
-        # Additional [] needed to keep m4 from mangling shell constructs.
-        [ GTEST_VERSION_2="`$GREP -A1 ^.C_INIT $GTEST_FRAMEWORK_SRC/configure.ac | $TAIL -n 1 | $SED -E -e 's/ +\[(.*)],/\1/'`" ]
-        if test "x$GTEST_VERSION_2" != "x1.8.0"; then
-          AC_MSG_ERROR([gtest at $GTEST_FRAMEWORK_SRC does not seem to be version 1.8.1 B])
+        if test "x$GTEST_VERSION_1" != "x1.13.0"; then
```
[the remark this answers is not in the document]
Comment (vidmik) [reply make/autoconf/lib-tests.m4]: Updated the logic to match that used for jtreg.

**Review by erikj79 — APPROVED** [review]

**Comment by vidmik** [comment]
Thank you @erikj79 & @magicus!

/integrate

**Pull request closed without merging**